### ISSUE TITLE
Objective-C Generics

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#87]
Since Xcode 7.0, it is allowed to have ObjC generics:

```
NSArray<MyClass *> *array;
```

This should be then translated into:

```
let array: [MyClass]
```

The same goes for `NSSet`, `NSOrderedSet` and their mutable counterparts.

Similarly, `NSDictionary<NSString *, id>` -> `[String:AnyObject]`.

**Comment by drkameleon** [comment]
Hmmm... I'll have to have a look into the official documentation. It makes sense. :)

**Comment by charlieMonroe** [comment]
Of course, you can make your own class with generics, but for now, at least these basic classes would be great :)

**Comment by drkameleon** [comment]
Issue (hopefully lol) fixed as of the upcoming 2.1.

P.S. That was a very tricky one.

----

**Example:**

```objective-c
@class MyClass;

@implementation aClass

-(void)someFunc:(NSArray *)a {
    NSArray<NSString*> *arr = @[ @"one", @"two", @"three" ];
    NSDictionary<NSString*, id> *dict = @{
        @"something" : arr 
    };
    NSArray<MyClass *> *array;

}

@end
```

**Output:**
```swift
class aClass {

	func someFunc(a: Array) {
		var arr: [String] = ["one","two","three"]
		var dict: [String:AnyObject] = ["something":arr]
		var array: [MyClass]
	
	}
	
}
```
